includes/class-platform-checkout-tracker.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Class Platform_Checkout_Tracker
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Automattic\Jetpack\Tracking;
+use WC_Payments_Features;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * Track Platform Checkout related events
+ */
+class Platform_Checkout_Tracker extends Tracking {
+
+	/**
+	 * Platform checkout event prefix
+	 *
+	 * @var string
+	 */
+	private static $prefix = 'woocommerceanalytics';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \WC_Payments_Http_Interface $http    A class implementing WC_Payments_Http_Interface.
+	 */
+	public function __construct( $http ) {
+		/**
+		 * Tracking class expects a Jetpac\Connection\Manager instance.
+		 * We pass in WC_Payments_Http_Interface which is a wrapper around the Connection Manager.
+		 *
+		 * @psalm-suppress InvalidArgument
+		 */
+		parent::__construct( self::$prefix, $http );
+		add_action( 'wp_ajax_platform_tracks', [ $this, 'ajax_tracks' ] );
+		add_action( 'wp_ajax_nopriv_platform_tracks', [ $this, 'ajax_tracks' ] );
+	}
+
+	/**
+	 * Override jetpack-tracking's ajax handling to use internal record_event method.
+	 */
+	public function ajax_tracks() {
+		// Check for nonce.
+		if (
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			empty( $_REQUEST['tracksNonce'] ) || ! wp_verify_nonce( $_REQUEST['tracksNonce'], 'platform_tracks_nonce' )
+		) {
+			wp_send_json_error(
+				__( 'You aren’t authorized to do that.', 'woocommerce-payments' ),
+				403
+			);
+		}
+
+		if ( ! isset( $_REQUEST['tracksEventName'] ) ) {
+			wp_send_json_error(
+				__( 'No valid event name or type.', 'woocommerce-payments' ),
+				403
+			);
+		}
+
+		$tracks_data = [];
+		if ( isset( $_REQUEST['tracksEventProp'] ) ) {
+			$event_prop = wc_clean( wp_unslash( $_REQUEST['tracksEventProp'] ) );
+			if ( is_array( $event_prop ) ) {
+				$tracks_data = $event_prop;
+			}
+		}
+
+		$this->record_event( sanitize_text_field( wp_unslash( $_REQUEST['tracksEventName'] ) ), $tracks_data );
+
+		wp_send_json_success();
+	}
+
+	/**
+	 * Generic method to track user events.
+	 *
+	 * @param string $event name of the event.
+	 * @param array  $data array of event properties.
+	 */
+	public function record_event( $event, $data = [] ) {
+
+		if ( ! $this->should_track() ) {
```

Comment:
When testing I was confused that adding to cart as a logged-in admin actually called `$this->record_event()`. 
I didn't notice at first that `record_event()` would short-circuit because of `! $this->should_track()`. While it is nice to have a single line of code that can break out of recording all events (avoiding duplication), it was not intuitive to me at first. 

I would have expected `track_add_to_cart()` to conditionally call `$this->record_event()` itself based on its own specific conditions. So if you were an admin it would return instead of calling `$this->record_event()`.
I know that `track_add_to_cart()` doing the conditional check will likely result in duplicated code but it seems like doing all of those checks in `record_event()` could end up becoming inflexible as you add more tracks.

An alternative is to go the WooCommerce way and call the method `maybe_record_event()` 😆

Reply:
Thanks for having a look @tommyshellberg. `should_track` contains the base conditions that should be checked for all platform-checkout track events. Having them defined in the `record_event` method will make it less susceptible to accidentally capturing these events when we should not. So I am opting towards having the `should_track` condition as-is. However, I agree that short-circuit logic within `record_event` method is confusing, and as you suggested, adding `maybe_` suffix might clarify things a bit. Updated in https://github.com/Automattic/woocommerce-payments/pull/3796/commits/4684915fae27d304ad56b05b1f2ee2f3ff827a7d. Thanks!